pkg/datastore/ethereum/database.go

```diff
@@ -0,0 +1,33 @@
+package ethereum
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethdb"
+
+	"github.com/vulcanize/vulcanizedb/pkg/datastore/ethereum/level"
+)
+
+type Database interface {
+	GetBlock(hash []byte, blockNumber int64) *types.Block
+	GetBlockHash(blockNumber int64) []byte
+	GetBlockReceipts(blockHash []byte, blockNumber int64) types.Receipts
+	GetHeadBlockNumber() int64
+}
+
+func CreateDatabase(config DatabaseConfig) (Database, error) {
+	switch config.Type {
+	case Level:
+		levelDBConnection, err := ethdb.NewLDBDatabase(config.Path, 128, 1024)
+		if err != nil {
+			return nil, err
+		}
+		levelDBReader := level.NewLevelDatabaseReader(levelDBConnection)
+		levelDB := level.NewLevelDatabase(levelDBReader)
+		return levelDB, nil
+	default:
+		return nil, errors.New(fmt.Sprintf("Unknown ethereum database: %s", config.Path))
```

Comment:
can use `fmt.Errorf('msg')` instead of `errors.New(fmt.Sprintf('msg'))`

Reply:
👍